api/proxy/store/builder/storage_manager_builder.go

```diff
@@ -282,6 +283,28 @@ func buildEigenDAV2Backend(
 	if err != nil {
 		return nil, fmt.Errorf("new cert verifier: %w", err)
 	}
+	if !isRouter {
+		// We call GetCertVersion to ensure that the cert verifier is of a supported version. See
+		// https://github.com/Layr-Labs/eigenda/blob/d0a14fa44/contracts/src/integrations/cert/interfaces/IVersionedEigenDACertVerifier.sol#L12
+		// https://github.com/Layr-Labs/eigenda/blob/d0a14fa44/contracts/src/integrations/cert/EigenDACertVerifier.sol#L79
+		// We pass in block 0 because a static certVerifierAddress provider is used when not using a router,
+		// so the block number is not relevant.
+		certVersion, err := certVerifier.GetCertVersion(ctx, 0)
+		if err != nil {
+			return nil, fmt.Errorf(
+				"failed to eth-call certVersion(), meaning that you either have network problems with your eth node, or "+
+					"%s is not a CertVerifier version >= V3, which is required by this version of proxy: %w",
+				routerOrImmutableVerifierAddr.Hex(), err)
+		}
+		// Note that we also support certV2s, just not V2 CertVerifiers.
+		// This is because we transform certV2s into certV3s and verified using the CertVerifierV3 contract.
+		// However, the serialization logic, as well as some functions needed during the dispersal path (eg. requiredQuorums),
+		// are only compatible/available with CertVerifier V3, hence the requirement here.
+		if certVersion != 3 {
```

Comment:
quick glance but why not?
```
certVersion >= 3
```

Reply:
Because certVersion should be treated as a major semver version (patrick is working on changing to an actual semver value). Then version 4+ means the interface change, so the proxy code/bindings is VERY likely not going to be compatible with the new cert version interface.

semver basically gives us protection from future incompatible changes (as long as our future selves are good enough to notice when the breaking change happens and to bump the major version). If we do that then old versions of proxy will start crashing on startup and force their users to update to more recent version that supports v4 for example. This is best of all words imo.